### PR TITLE
support for local.directory.sequence.id

### PR DESCRIPTION
**Pull request by mr-agrwal**
There is support for the same using PinotSparkJobLauncher. But not using the LaunchDataIngestionJobCommand.

### CODE REVIEW

File: pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark/src/main/java/org/apache/pinot/plugin/ingestion/batch/spark/SparkSegmentGenerationJobRunner.java [hunk 1]
```diff
@@ -29,12 +30,10 @@
...
 import java.net.URI;
 import java.nio.file.FileSystems;
+import java.nio.file.Path;
 import java.nio.file.PathMatcher;
 import java.nio.file.Paths;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.UUID;
+import java.util.*;
```
Comment (fx19880617) [inline pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark/src/main/java/org/apache/pinot/plugin/ingestion/batch/spark/SparkSegmentGenerationJobRunner.java:36]: expend the .*

File: pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark/src/main/java/org/apache/pinot/plugin/ingestion/batch/spark/SparkSegmentGenerationJobRunner.java [hunk 2]
```diff
@@ -200,8 +199,30 @@ public void run()
...
+        Map<String, List<String>> localDirIndex = new HashMap<>();
+        for (String filteredFile : filteredFiles) {
+          Path filteredParentPath = Paths.get(filteredFile).getParent();
+          if (!localDirIndex.containsKey(filteredParentPath.toString())) {
+            localDirIndex.put(filteredParentPath.toString(), new ArrayList<>());
+          }
+          localDirIndex.get(filteredParentPath.toString()).add(filteredFile);
+        }
+        for (String parentPath: localDirIndex.keySet()){
+          List<String> siblingFiles = localDirIndex.get(parentPath);
```
Comment (fx19880617) [inline pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-spark/src/main/java/org/apache/pinot/plugin/ingestion/batch/spark/SparkSegmentGenerationJobRunner.java:217]: suggest to sort siblingFiles list, so rerun segment creation job will give same segments list

**Comment by fx19880617** [comment]
Can you also update the PR description of why we need to do this, e.g. multiple input files per day and you want to refresh data on same day ?
And give an example of input data directory structure and expected segments directory structure.

**Comment by kishoreg** [comment]
please update the docs as well

**Comment by fx19880617** [comment]
Please remember to update the doc here: https://docs.pinot.apache.org/basics/components/segment#load-data-in-batch

**Pull request merged**